src/components/repl/Repl.tsx

```diff
@@ -0,0 +1,235 @@
+import { FC, useEffect, useState } from 'react';
+
+import { PyWorker, PyodideStatus } from '@graasp/pyodide';
+
+import { Alert, Stack } from '@mui/material';
+import Grid from '@mui/material/Unstable_Grid2';
+
+import { CODE_EXECUTION_SETTINGS_NAME } from '../../config/appSettingsTypes';
+import { REPL_CONTAINER_CY, REPL_EDITOR_ID_CY } from '../../config/selectors';
+import { DEFAULT_CODE_EXECUTION_SETTINGS } from '../../config/settings';
+import { CodeVersionType } from '../../interfaces/codeVersions';
+import { CodeExecutionSettingsKeys } from '../../interfaces/settings';
+import { useAppDataContext } from '../context/AppDataContext';
+import { useSettings } from '../context/SettingsContext';
+import CodeEditor from './CodeEditor';
+import InputArea from './InputArea';
+import ReplToolbar from './ReplToolbar';
+import ShowFigures from './ShowFigures';
+
+type Props = {
+  seedValue: CodeVersionType;
+  // todo: implement "bo-back" button
+  // eslint-disable-next-line react/no-unused-prop-types
+  onClose: () => void;
+};
+
+const Repl: FC<Props> = ({ seedValue }) => {
+  const [worker, setWorker] = useState<PyWorker | null>(null);
+  const [output, setOutput] = useState<string>('');
+  const [prompt, setPrompt] = useState<string>('');
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [error, setError] = useState<string | null>(null);
+  const [figures, setFigures] = useState<string[]>([]);
+
+  const { codeAppData } = useAppDataContext();
+  // sort app data by the latest to the oldest
+  const sortedCodeVersions = codeAppData.sort((a, b) =>
+    Date.parse(a.updatedAt) > Date.parse(b.updatedAt) ? 1 : -1,
+  );
+  const latestCode = sortedCodeVersions.get(0)?.data?.code || '';
+
+  const currentCode = seedValue ? seedValue.code : latestCode;
+
+  // todo: get value from app data for the user
+  const [value, setValue] = useState(currentCode);
+
+  const {
+    [CODE_EXECUTION_SETTINGS_NAME]:
+      codeExecSettings = DEFAULT_CODE_EXECUTION_SETTINGS,
+  } = useSettings();
+
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [isWaitingForInput, setIsWaitingForInput] = useState(false);
+  const [replStatus, setReplStatus] = useState<PyodideStatus>(
+    PyodideStatus.LOADING_PYODIDE,
+  );
+
+  // register worker on mount
+  useEffect(() => {
+    // todo: reconciliate the concat output option
+    const workerInstance = new PyWorker(
+      'https://spaenleh.github.io/graasp-pyodide/fullWorker.min.js',
```

Comment:
Factor out?

Reply:
Maybe it should be an input from builder? not sure yet.

Reply:
I am not sure how this would be done as an input from builder ? Maybe it could be an env var passed during the build process.